ArduCopter/control_autonomous.cpp

```diff
@@ -206,7 +206,7 @@ bool Copter::autonomous_controller(float &target_climb_rate, float &target_roll,
 
     static States state = HOLD_CENTER;
 
-    static int stateCounter = 0;
+    static int stateCounter = 0;    //CY: wouldn't this set it to 0 every iter of the 'while loop' ?
```

Comment:
I don't believe so, the variable static so it is shared between all function calls. At line 313, they use the same thing for printing at an interval.